Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -141,6 +139,11 @@ - (void)startWithOptions:(SentryReplayOptions *)replayOptions
                                                    name:UIApplicationDidEnterBackgroundNotification
                                                  object:nil];
 
+        [NSNotificationCenter.defaultCenter addObserver:self
```

Comment:
`m`: Again, I'm not a fan of using the NSNotificationCenter directly as it cause multiple problems with testing in the past. I would prefer to use `SentryNSNotificationCenterWrapper`.

Reply:
Technically, there is no reason for NSNotificationCenter to cause any problems. 
Maybe the old flakiness was due to issues in the test. 
I will monitor this to ensure it does not become flaky.

Reply:
No clue. It's just what I experienced in the past.